src/robot/errors.py

```diff
@@ -53,6 +53,14 @@ class DataError(RobotError):
     this exception with care.
     """
 
+class VariableError(DataError):
+    """Used when variable syntax or evaluation is invalid.
+
+    VariableErrors are caught by keywords that run other keywords
+    (e.g. `Run Keyword And Expect Error`). Libraries should thus use
+    this exception with care.
```

Comment:
I think the last sentence can be removed. Libraries should never use these exceptions. If we want to make that more explicit, it could be documented either on module level or in RobotError base class.